internal/pkg/limit/limiter.go

```diff
@@ -63,11 +113,30 @@ func (l *Limiter) Acquire() (ReleaseFunc, error) {
 	return releaseFunc, nil
 }
 
-func (l *Limiter) release() {
+func (l *limiter) release() {
 	if l.maxLimit != nil {
 		l.maxLimit.Release(1)
 	}
 }
 
+func (l *limiter) wrap(logger zerolog.Logger, level zerolog.Level, h httprouter.Handle, i StatIncer) httprouter.Handle {
+	return func(w http.ResponseWriter, r *http.Request, p httprouter.Params) {
+		dfunc := i.IncStart()
```

Comment:
For better or worse, this logic is different than the original implementation.  In the original implementation, the counter was not incremented until:

1) The limiter semaphore was acquired
2) The transaction was authorized

Reply:
Yeah, i made a comment about that, we can definitely restore that behaviour if we need to.
But with it tracking every attempt it would mean that the error count associated with a route can't be higher than the total calls

Reply:
Just metrics; doesn't affect logic so doesn't matter much.